src/mo_frontend/typing.ml

```diff
@@ -3588,17 +3647,27 @@ and gather_dec env scope dec : Scope.t =
     let open Scope in
     if T.Env.mem id.it scope.val_env then
       error_duplicate env "" id;
-    let scope' = gather_block_decs env decs in
+    (* NOTE: This is a bit ugly. We need to extend the environment
+       with any mixin imports we've found so far, so Include's in the
+       actor body can be resolved *)
+    let scope' = gather_block_decs { env with mixins = T.Env.adjoin scope.mixin_env env.mixins } decs in
     let ve' = add_id scope.val_env id (object_of_scope env obj_sort.it dec_fields scope' at) in
     let obj_env = T.Env.add id.it scope' scope.obj_env in
     { val_env = ve';
       typ_env = scope.typ_env;
       lib_env = scope.lib_env;
       con_env = scope.con_env;
       obj_env = obj_env;
+      mixin_env = scope.mixin_env;
       fld_src_env = scope.fld_src_env;
     }
-  | LetD (pat, _, _) -> gather_pat env scope pat
+  | LetD (pat, exp, _) -> (match is_mixin_import env exp.it with
+    | None -> gather_pat env scope pat
+    | Some (imports, args, t, decs) ->
+      match pat.it with
+      | VarP id -> Scope.adjoin scope (Scope.mixin id.it (imports, args, t, decs))
```

Comment:
So I guess this is a bit dubious because we are confusing the name space of values with the name space of mixins....but I can live with that.